### PR TITLE
Add `convert_lit_lora_checkpoint.py`

### LINKED ISSUE #374: Converting models to original format after finetuning wit LoRA and QLoRA
User raised issued regarding KeyError after finetuning with LoRA in the following comment: https://github.com/Lightning-AI/lit-gpt/issues/355#issuecomment-1668854378

### LINKED ISSUE #483: Save the model config in the training scripts
while testing conversion of LoRA finetuned checkpoints to their original format, it was found that having a `lora_config.json` saved to `out/lora/` would be beneficial when creating a new model with `model = GPT.from_name(model_name, **lora_config_during_finetuning)` or `model = GPT(**lora_config_during_finetuning)`.

intended changes in `finetune.lora.py` are:

```python
import json
from dataclasses import asdict

def save_lora_config(fabric, model, out_dir: Path):
    fabric.print(f"Saving LoRA config to {str(out_dir)!r}")
    config_payload = {k: v for k, v in asdict(model.config).items() if k != "org"}
    with open(out_dir / "lit_lora_config.json", "w") as config_path:
        json.dump({"config": config_payload}, config_path, indent=4))
```

example output

```json
{
    "config": {
        "name": "Llama-2-7b-hf",
        "block_size": 1,
        "vocab_size": 1,
        "padding_multiple": 64,
        "padded_vocab_size": 64,
        "n_layer": 1,
        "n_head": 4,
        "n_embd": 8,
        "rotary_percentage": 1.0,
        "parallel_residual": false,
        "bias": false,
        "n_query_groups": 4,
        "shared_attention_norm": false,
        "_norm_class": "RMSNorm",
        "norm_eps": 1e-05,
        "_mlp_class": "LLaMAMLP",
        "intermediate_size": 11008,
        "condense_ratio": 1,
        "r": 1,
        "alpha": 8,
        "dropout": 0.1,
        "to_query": true,
        "to_key": false,
        "to_value": true,
        "to_projection": false,
        "to_mlp": false,
        "to_head": false
    }
}
```

### PR DESCRIPTION
**Pull request by JustinGoheen**
Adds `scripts/convert_lit_lora_checkpoint.py` to separate LoRA merging and conversion operations away from `convert_lit_checkpoint.py`.

closes #374 
closes #483

### CODE REVIEW

**Review by JustinGoheen — COMMENTED** [review]
need to fix out_dir for full finetuning

**Comment by JustinGoheen** [comment]
Blocked by #364

File: tutorials/convert_lit_models.md
Comment (carmocca) [inline tutorials/convert_lit_models.md:24]: As mentioned in other issues/PRs, we don't support adapter exports so this whole section should be removed with an explanation of this limitation

File: tutorials/convert_lit_models.md
Comment (carmocca) [inline tutorials/convert_lit_models.md:59]: Doesn't this apply to all sorts of issues?

Comment (JustinGoheen) [reply tutorials/convert_lit_models.md]: removed in [7cbea36](https://github.com/Lightning-AI/lit-gpt/pull/362/commits/7cbea36843faf324908441d6fc08faf86e39259b)

File: tutorials/convert_lit_models.md
Comment (JustinGoheen) [reply tutorials/convert_lit_models.md]: updated in [7cbea36](https://github.com/Lightning-AI/lit-gpt/pull/362/commits/7cbea36843faf324908441d6fc08faf86e39259b)

File: tutorials/convert_lit_models.md
Comment (carmocca) [inline tutorials/convert_lit_models.md:36]:
```suggestion
    --checkpoint_name lit_model_lora_finetuned.pth \
```

**Review by carmocca — DISMISSED** [review]

File: tutorials/convert_lit_models.md
Comment (JustinGoheen) [reply tutorials/convert_lit_models.md:32]: Are you cool with me adding this into convert_lit_checkpoint? This seems like something I should enable in the script, instead of leaving it to the users.

Comment (JustinGoheen) [reply tutorials/convert_lit_models.md:32]: if yes, I will rename this PR accordingly.

File: tutorials/convert_lit_models.md
Comment (JustinGoheen) [reply tutorials/convert_lit_models.md]: fixed in [5d87fa9](https://github.com/Lightning-AI/lit-gpt/pull/362/commits/5d87fa924e8ac4e2932855bfb6e9c8eedf91e65f)

File: tutorials/convert_lit_models.md
Comment (JustinGoheen) [reply tutorials/convert_lit_models.md:45]: implemented in [5d87fa9](https://github.com/Lightning-AI/lit-gpt/pull/362/commits/5d87fa924e8ac4e2932855bfb6e9c8eedf91e65f)

File: scripts/convert_lit_checkpoint.py
Comment (carmocca) [inline scripts/convert_lit_checkpoint.py:218]: Before merging, you need to load 2 checkpoints, the pretrained weights and the lora weights. See for example how the generation script does it https://github.com/Lightning-AI/lit-gpt/blob/main/generate/lora.py#L110-L112

File: tests/test_convert_lit_checkpoint.py
Comment (carmocca) [inline tests/test_convert_lit_checkpoint.py:214]: Simply unit-testing `merge_lora_weights_for_conversion` would be enough to test the changes in this PR. This test is unnecessarily complex, it doesn't need to train a model

**Comment by JustinGoheen** [comment]
working on separating LoRA conversion into a new module: `convert_lit_lora_checkpoint.py`.

File: scripts/convert_lit_lora_checkpoint.py
Comment (carmocca) [inline scripts/convert_lit_lora_checkpoint.py:1]: You can now use https://github.com/Lightning-AI/lit-gpt/blob/main/scripts/merge_lora.py since #495 has landed

File: scripts/merge_lora.py
Comment (JustinGoheen) [inline scripts/merge_lora.py:68]: if strict is set to`True`, and if `lora_A` and `lora_B` weights exist, they will cause an error here.

**Comment by carmocca** [comment]
Closing this PR in favor of #572. You could still open a new one to address #483 though, if you'd like

**Pull request closed without merging**
